### PR TITLE
ESLint: Re-enable rule default-props-match-prop-types

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
Re-enable ESLint rule `default-props-match-prop-types`, which was disabled in PR #10839. Code was refactored to fix the errors raised by the rule.

### TEST PLAN
Run `npm run lint`, verify that there are no new Javascript/Typescript errors.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/dashboard/components/ColorSchemeControlWrapper.jsx [hunk 1]
```diff
@@ -30,7 +30,6 @@ const propTypes = {
 
 const defaultProps = {
   colorScheme: undefined,
-  onChange: () => {},
```
Comment (mistercrunch) [inline superset-frontend/src/dashboard/components/ColorSchemeControlWrapper.jsx:33]: this one is making me nervous, can we remove the `.isRequired` instead?

File: superset-frontend/src/dashboard/components/DashboardBuilder.jsx [hunk 2]
```diff
@@ -64,7 +64,6 @@ const propTypes = {
 };
 
 const defaultProps = {
-  showBuilderPane: false,
```
Comment (mistercrunch) [inline superset-frontend/src/dashboard/components/DashboardBuilder.jsx:67]: seems less risky to remove the `.isRequired` here too  (unless you can confirm you checked all the calls)

File: superset-frontend/src/dashboard/components/SliceAdder.jsx [hunk 3]
```diff
@@ -45,7 +45,6 @@ const propTypes = {
 };
 
 const defaultProps = {
-  selectedSliceIds: [],
```
Comment (mistercrunch) [inline superset-frontend/src/dashboard/components/SliceAdder.jsx:48]: let's remove `.isRequired` here to (unless you can confirm you checked all the calls)

**Review by mistercrunch — CHANGES_REQUESTED** [review]

File: superset-frontend/src/dashboard/components/ColorSchemeControlWrapper.jsx (hunk 1 above)
Comment (kgabryje) [reply superset-frontend/src/dashboard/components/ColorSchemeControlWrapper.jsx:33]: 👍

File: superset-frontend/src/dashboard/components/DashboardBuilder.jsx (hunk 2 above)
Comment (kgabryje) [reply superset-frontend/src/dashboard/components/DashboardBuilder.jsx:67]: I did, but you're right, let's keep the default and remove isRequired

File: superset-frontend/src/dashboard/components/SliceAdder.jsx (hunk 3 above)
Comment (kgabryje) [reply superset-frontend/src/dashboard/components/SliceAdder.jsx:48]: I did, but you're right, let's keep the default and remove isRequired

File: superset-frontend/src/dashboard/components/DashboardBuilder.jsx (hunk 2 above)
Comment (mistercrunch) [reply superset-frontend/src/dashboard/components/DashboardBuilder.jsx:67]: I was confused about the type here and tracked it into the child component, and it looks unused there. Can you double check and remove all references to it here if that's the case?

Comment (kgabryje) [reply superset-frontend/src/dashboard/components/DashboardBuilder.jsx:67]: It seems that not only `showBuilderPane` was unused, but also `colorScheme` and `setColorSchemeAndUnsavedChanges`. I removed references from this file, containers/DashboardBuilder and the spec file.

**Review by rusackas — APPROVED** [review]
LGTM, but will leave merging to @mistercrunch to make sure he's happy with the changes made :)

**Comment by kgabryje** [comment]
@mistercrunch Can we merge it now? I made the changes that you asked for

**Comment by graceguo-supercat** [comment]
@kgabryje we just found this PR broke Dashboard tab switch:
mSqaIe8eS4

i can't find exactly which line of change is wrong. But since dashboard tab switch is very critical function, i will have to revert this PR.

**Comment by kgabryje** [comment]
@graceguo-supercat I'm sorry to hear that! I'll try to look into it

**Pull request merged**